packages/pipeline/src/data_sources/slippage/index.ts

```diff
@@ -0,0 +1,55 @@
+import { fetchAsync } from '@0x/utils';
+
+const EDPS_BASE_URL = 'https://ethereum-dex-prices-service.production.airswap.io';
+const PRICE_BASE_URL = 'https://min-api.cryptocompare.com/data/price?tsyms=USD';
+
+export type EdpsResponse = EdpsWrapper[];
+
+export interface EdpsWrapper {
+    [key: string]: EdpsExchange;
+}
+
+export interface EdpsExchange {
+    exchangeName: string;
+    totalPrice: number;
+    tokenAmount: number;
+    tokenSymbol: string;
+    avgPrice: number;
+    timestamp: number;
+    error: string;
+}
+
+export interface PriceResponse {
+    USD: number;
+}
+
+// tslint:disable:prefer-function-over-method
+// ^ Keep consistency with other sources and help logical organization
+export class EdpsSource {
+    /**
+     * Call Ethereum DEX Price Service API.
+     */
+    public async getEdpsAsync(direction: string, symbol: string, amount: number): Promise<Map<string, EdpsExchange>> {
```

Comment:
So this is actually the first use of es6 `Map` in the entire codebase! While I don't think we have particularly strong reasons for not using `Map`, we do always just use simple Objects and it might be worth switching this just to stay consistent. We do have an `ObjectMap` type (`import { ObjectMap } from '@0x/types';`) that we use pretty often, that is just like your `EdpsWrapper` type, but with some generics.

https://medium.com/front-end-weekly/es6-map-vs-object-what-and-when-b80621932373

Reply:
Oh I was wrong about this. We have used `Map` before, but never returned one from a public interface.

Reply:
Great input - thanks, will change it